### PR TITLE
RDoc-1225 - Add EnsureDatabaseExists example to create database

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/server-wide/create-database.dotnet.markdown [hunk 1]
```diff
@@ -53,6 +53,17 @@ If `Topology` is specified, the `replicationFactor` will be ignored.
 {CODE-TAB:csharp:Async CreateDatabaseAsync@ClientApi\Operations\Server\CreateDeleteDatabase.cs /}
 {CODE-TABS/}
 
+{INFO:Information}
+To ensure database exists before creating it we can use the following example
+
+###Exapmle I - EnsureDatabaseExists
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/server-wide/create-database.dotnet.markdown]: Exapmle  -> Example

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/server-wide/create-database.dotnet.markdown [hunk 2]
```diff
@@ -53,6 +53,17 @@ If `Topology` is specified, the `replicationFactor` will be ignored.
...
 {CODE-TABS/}
 
+{INFO:Information}
+To ensure database exists before creating it we can use the following example
+
+###Exapmle I - EnsureDatabaseExists
+{CODE-TABS}
+{CODE-TAB:csharp:Sync EnsureDatabaseExists@ClientApi\Operations\Server\CreateDeleteDatabase.cs /}
+{CODE-TAB:csharp:Async EnsureDatabaseExistsAsync@ClientApi\Operations\Server\CreateDeleteDatabase.cs /}
+{CODE-TABS/}
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/server-wide/create-database.dotnet.markdown]: We should add a note that creation of a database requires admin certificate

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs [hunk 3]
```diff
@@ -50,6 +53,62 @@ public class CreateDatabaseOperation
...
+
+            if (string.IsNullOrWhiteSpace(database))
+                return;
+
+            try
+            {
+                store.Maintenance.ForDatabase(database).Send(new GetStatisticsOperation());
+            }
+            catch (DatabaseDoesNotExistException)
+            {
```
Comment (arekpalinski) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: If db doesn't exist and `createDatabaseIfNotExists == false` you aren't throwing

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs [hunk 4]
```diff
@@ -50,6 +53,62 @@ public class CreateDatabaseOperation
...
+                if (createDatabaseIfNotExists)
+                {
+                    try
+                    {
+                        store.Maintenance.Server.Send(new CreateDatabaseOperation(new DatabaseRecord(database)));
+                    }
+                    catch (ConcurrencyException)
+                    {
```
Comment (arekpalinski) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: let's add a comment that a db was created meanwhile

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs [hunk 5]
```diff
@@ -50,6 +53,62 @@ public class CreateDatabaseOperation
             }
         }
 
+        #region EnsureDatabaseExists
+        public void EnsureDatabaseExists(IDocumentStore store, string database = null, bool createDatabaseIfNotExists = true)
+        {
+            database = database ?? store.Database;
+
+            if (string.IsNullOrWhiteSpace(database))
+                return;
```
Comment (arekpalinski) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: We should throw ArgumentException here. We have invalid db name

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs (hunk 3 above)
Comment (arekpalinski) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: You can add additional parameter `bool ignoreFailures = false` to control if it should throw or now

Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: It is just an example that the user can use to ensure if the database exists, it is up to him if he wants to throw something.
I Think that a commit here will suffice

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs (hunk 4 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: Meanwhile what we are using the CreateDatabaseOperation is not enough?

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs (hunk 5 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: You are right will change it

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/server-wide/create-database.dotnet.markdown (hunk 2 above)
Comment (IdanHaim) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/server-wide/create-database.dotnet.markdown]: Will add this

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/server-wide/create-database.dotnet.markdown (hunk 1 above)
Comment (IdanHaim) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/server-wide/create-database.dotnet.markdown]: Done

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs (hunk 3 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: Ok Added

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs (hunk 4 above)
Comment (IdanHaim) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/CreateDeleteDatabase.cs]: Done

**Pull request merged**